packages/app-mobile/components/screens/status.tsx

```diff
@@ -1,37 +1,49 @@
-const React = require('react');
-
-const { View, Text, Button, FlatList } = require('react-native');
-const Setting = require('@joplin/lib/models/Setting').default;
-const { connect } = require('react-redux');
-const { ScreenHeader } = require('../ScreenHeader');
-const ReportService = require('@joplin/lib/services/ReportService').default;
-const { _ } = require('@joplin/lib/locale');
-const { BaseScreenComponent } = require('../base-screen');
-const { themeStyle } = require('../global-style');
-
-class StatusScreenComponent extends BaseScreenComponent {
-	static navigationOptions() {
+import * as React from 'react';
+
+import { View, Text, Button, FlatList, TextStyle } from 'react-native';
+import Setting from '@joplin/lib/models/Setting';
+import { connect } from 'react-redux';
+import { ScreenHeader } from '../ScreenHeader';
+import ReportService, { ReportSection } from '@joplin/lib/services/ReportService';
+import { _ } from '@joplin/lib/locale';
+import { BaseScreenComponent } from '../base-screen';
+import { themeStyle } from '../global-style';
+import { AppState } from '../../utils/types';
+import checkDisabledSyncItemsNotification from '@joplin/lib/services/synchronizer/utils/checkDisabledSyncItemsNotification';
+import { Dispatch } from 'redux';
+
+interface Props {
+	themeId: number;
+	dispatch: Dispatch;
+}
+
+interface State {
+	report: ReportSection[];
+}
+
+class StatusScreenComponent extends BaseScreenComponent<Props, State> {
+	public static navigationOptions(): any {
 		return { header: null };
 	}
 
-	constructor() {
-		super();
+	public constructor(props: Props) {
+		super(props);
 		this.state = {
 			report: [],
 		};
 	}
 
-	UNSAFE_componentWillMount() {
-		this.resfreshScreen();
+	public override componentDidMount() {
```

Comment:
In this case, `componentDidMount` should be roughly equivalent to `componentWillMount`, because `refreshScreen` is async ([see the RFC that deprecated `componentWillMount`](https://github.com/reactjs/rfcs/blob/main/text/0006-static-lifecycle-methods.md#prefetching-async-data-during-mount)).